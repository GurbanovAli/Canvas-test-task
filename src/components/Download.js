import React from 'react';
import {
   useAppState
} from './ControlFile';

function Download() {
   const {
      url: href
   } = useAppState();
   const Component = href ? 'a' : 'button';

   return (
      <div className = 'download_button' >
      <Component
      className = 'button'
      href = { href }
      download = { href ? 'output' : null } >
      Download canvas
      </Component>
      </div>
   );
}

export default Download;
