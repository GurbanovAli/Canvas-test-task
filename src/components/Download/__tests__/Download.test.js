import React from 'react';
import { render } from '@testing-library/react';
import Download from '../Download';
import Control from '../../ControlFile';

test('<Download /> spec', () => {
    const { container } = render(
      <Control>
        <Download />
      </Control>
    );
    expect(container).toMatchSnapshot();
  });
