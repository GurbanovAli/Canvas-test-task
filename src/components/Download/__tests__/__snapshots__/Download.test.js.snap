// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div>
   
  <div
    class="download_button"
  >
    <button
      class="button"
    >
      Download canvas
    </button>
  </div>
   
</div>
`;

exports[`<Download /> spec 1`] = `
<div>
   
  <div
    class="download_button"
  >
    <button
      class="button"
    >
      Download canvas
    </button>
  </div>
   
</div>
`;

exports[`<Download /> spec renders the component 1`] = `
<div>
   
  <div
    class="download_button"
  >
    <button
      class="button"
    >
      Download canvas
    </button>
  </div>
   
</div>
`;
